Jasmine/Common/Models/Coordinate.swift

```diff
@@ -0,0 +1,29 @@
+import Foundation
+
+/// Defines a coordinate system that is zero-indexed, where the top left corner has the coordinate
+/// row = 0, col = 0.
+///
+/// - Author: Wang Xien Dong
+struct Coordinate {
+
+    /// Defines the origin of the coordinate system, which is zero.
+    static let zero = Coordinate(row: 0, col: 0)
+
+    let row: Int
+    let col: Int
+}
+
+extension Coordinate: Hashable {
+    /// Returns the hash value of this coordinate data structure.
+    var hashValue: Int {
+        return row + col * 31
```

Comment:
Use `row &+ col &* 31` to prevent overflowing.

Or rather, convert this to a unicode String of 2 characters and get their hash value. I would think using Swift's hashValue functions are better